### PR TITLE
fix(input-selection): fixed recursive use of CSL object

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

1. The CML WASM module is throwing "Error: recursive use of an object detected which would lead to unsafe aliasing in rust", This can happen when two segments of code try to borrow the same object.
The function that is throwing is using a constant which is a CML object of type BigNum, this object is then reused everytime the function is called.
2. We suspect that asm.js variant of CML is sometimes going out of memory.

# Proposed Solution

1. Create a new managed BigNum on every call and pass it to the Rust function.
2. Update test extension to use WASM variant of CML. As a positive consequence we managed to reduce bundle sizes by allowing webpack to do tree-shaking.

### CODE REVIEW

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Good catch @AngelCastilloB. Can you add a test to cover the bug being fixed? Also, I wonder if there are other cases of this bug. Have you done a more widespread review?

**Review by rhyslbw — CHANGES_REQUESTED** [review]
How about converting the `cmlUtil.maxBigNum` into a function instead (which makes it a real utility)? It would need the scope passed in.

File: packages/wallet/test/SingleAddressWallet/load.test.ts [hunk 1]
```diff
@@ -165,6 +165,82 @@ const assertWalletProperties2 = async (wallet: ObservableWallet) => {
...
+  return utxos;
+};
+
+/**
+ * Gets the asset list and the total amount of lovelace on a utxo set.
+ *
+ * @param utxos The utxos.
+ * @returns The total lovelace and the aggregated asset set.
+ */
+const getAssets = (utxos: Cardano.Utxo[]) => {
```
Comment (rhyslbw) [inline packages/wallet/test/SingleAddressWallet/load.test.ts:228]:
```suggestion
const getAssetsFromUtxos = (utxos: Cardano.Utxo[]) => {
```

Better if this was sourced in the `util` package

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/wallet/test/SingleAddressWallet/load.test.ts (hunk 1 above)
Comment (rhyslbw) [reply packages/wallet/test/SingleAddressWallet/load.test.ts:228]: This is actually not possible due to the current dependency `core` has on `util`, so disregard this suggestion. We can look at hoisting when the opportunity arises.

**Review by rhyslbw — DISMISSED** [review]

**Review by mkazlauskas — DISMISSED** [review]
Nice work!

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**